pkg/telemetry/gauge.go

```diff
@@ -54,6 +56,8 @@ func NewGaugeWithOpts(subsystem, name string, tags []string, help string, opts O
 			tags,
 		),
 	}
-	telemetryRegistry.MustRegister(g.pg)
+	if err := telemetryRegistry.Register(g.pg); err != nil {
```

Comment:
unit and integration tests fail because of `MustRegister` panics, I believe it's fair to use `Register` and log the error  instead as telemetry and this is not that critical to allow it cause an agent panic IMHO.
cc @remeh

Reply:
What's the reason for the CI to panic in your PR? If by any chance you have a link that you can send me offline.
The idea was exactly to panic as soon as possible for this (during the development phase, as you had) because the reason is most of the time valid, but I don't mind that much switching to an error log at startup, it is just less visible and the developer has to be more cautious.
However, I think that an issue here is that you can't be 100% sure that the logger is properly instanciated and configured. If I'm right, you may have to rely on `fmt` and not our logger.

Reply:
@remeh 
I think we have already exchanged in the past about my concerne using the global variable for accessing the registry.
it makes it difficult to unit-test some function that generate metrics.

Reply:
Thanks @clamoriniere , I share the concerns.
@remeh I changed the main logic of the PR and as a result unit tests are no longer failing but here is an example of the integration tests failing https://app.circleci.com/pipelines/github/DataDog/datadog-agent/3545/workflows/f3bb9c72-13fd-4568-bdb3-2edece42a292/jobs/235565/steps
I haven't looked (yet) into the integration tests code yet, but it's obvious `test panicked: duplicate metrics collector registration attempted`

Reply:
If we want to keep the current design of the telemetry pkg we have at least to expose a `Reset()` function to reset the singleton when needed by tests.

Reply:
Let me know if the current change is acceptable, I've put back `MustRegister` and added a `Reset()` function.

Reply:
Got it.
@clamoriniere I remember the conversation, and if I recall properly that is when I've introduced "our" global instance instead of using the one from the prometheus package: https://github.com/DataDog/datadog-agent/blob/master/pkg/telemetry/prom_telemetry.go (this whole file was not existing before our conversation), following this, I think that this `Reset()` addition is 👌 
I'm not against completely moving to a local instance (removing the global), it is just that it would have to been passed through all structs needing it (and future features), hardening the telemetry adoption which is already an optional step. Thanks for the change @ahmed-mez, that looks good to me!